src/prog_models/data_models/lstm_model.py

```diff
@@ -126,23 +126,22 @@ def summary(self, file= sys.stdout, expand_nested=False, show_trainable=False):
         self.model.summary(print_fn= file.write, expand_nested = expand_nested, show_trainable = show_trainable)
         
     @staticmethod
-    def pre_process_data(data, window, **kwargs):
+    def pre_process_data(inputs, outputs, window, **kwargs):
         """
         Pre-process data for the LSTMStateTransitionModel. This is run inside from_data to convert the data into the desired format 
 
         Args:
-            data (List[Tuple] or Tuple (where Tuple is equivilant to [Tuple]))): Data to be processed. each element is of format (input, output), where input and output can be ndarray or SimulationResult
+            inputs (List[ndarray or SimulationResult]): Data to be processed. Each element is of format, ndarray or SimulationResult
+            outputs (List[ndarray or SimulationResult]): Data to be processed. Each element is of format, ndarray or SimulationResult
             window (int): Length of a single sequence
 
         Returns:
             Tuple[ndarray, ndarray]: pre-processed data (input, output). Where input is of size (num_sequences, window, num_inputs) and output is of size (num_sequences, num_outputs)
         """
-        # Data is a List[Tuple] or Tuple (where Tuple is equivilant to [Tuple]))
-        # Tuple is (input, output)
 
         u_all = []
         z_all = []
-        for (u, z) in data:
+        for (u, z) in zip(inputs, outputs):
```

Comment:
Do we need to make sure inputs and outputs are the same size, or will zip do that for us? All sizes moving forward are defined by the size of inputs only